python/ray/train/v2/_internal/execution/checkpoint/checkpoint_manager.py

```diff
@@ -154,25 +161,25 @@ def update_checkpoints_with_metrics(
                 continue
             checkpoint_result = self._pending_training_results[checkpoint]
             checkpoint_result.metrics.update(metrics)
-            try:
-                self._checkpoint_results.remove(checkpoint_result)
-            except ValueError:
-                logger.warning(
-                    f"Checkpoint {checkpoint} not found in checkpoint results. "
+            if checkpoint_result not in self._checkpoint_results:
+                raise ValueError(
+                    f"Checkpoint {checkpoint} was in pending training results but not "
+                    "checkpoint results. "
                 )
-                continue
+            self._checkpoint_results.remove(checkpoint_result)
             _insert_into_sorted_list(
                 self._checkpoint_results,
                 checkpoint_result,
                 key=self._get_checkpoint_score,
+                checkpoint_to_report_index=self._checkpoint_to_report_index,
```

Comment:
possible to just incorporate the checkpoint index as the default checkpoint score in `get_checkpoint_score`?

Reply:
oh I see your comment. you wanted to minimize train v1 changes?